### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by dantman** [material-native/material-native#31]
The library will need good documentation before release.

The following libraries may be useful in generating automatic documentation for components:

* https://github.com/reactjs/react-docgen
* https://www.npmjs.com/package/react-doc-generator
* https://github.com/adroll/gulp-react-docs

Depending on the output and size documentation can be put in one of multiple locations:
* Merged into the README.md
* Saved into a separate .md file in the repo and linked to on GitHub
* Published to a GitHub pages site
* https://readthedocs.org/
* https://github.com/integrations/gitbook

We'll also need some documentation pages for things like how to use MaterialTheme.

**Comment by Leva7** [comment]
I wanna give it a shot. Here's a [rough draft](https://gist.github.com/Leva7/56e3f240429cc7197e946008aa88f8ca) of documentation for the Avatar component (don't mind the question marks and factual mistakes). The outline was built with the `react-doc-generator` tool, then I worked on it manually. 
How do you like it? Should I continue? Is there anything you want to change about the formatting?

**Comment by dantman** [comment]
The overall layout could be a good base (title, info, example, props table).

However I don't want any manual post-processing to be involved in this. Eventually the code is going to use flow, so we'll have proper types for the docs. And I plan to have descriptions and examples as comments in the code.

I'm holding off on the docs for now because there are some other things that should be done before making them and I anticipate that sometime in the future working on the documentation may be something a client I have will ask me for.

**Comment by Leva7** [comment]
Yeah, I get the automatic generation part. It's just the fact that those comments in the code have to be actually written, so I figured it'd be better to write them in a separate markdown file and then update the code accordingly. 

However, there are problems (at least with the way `react-doc-generator` parsed your source and it uses `react-docgen` under the hood). The generated documentation contained a table with only two props - the ones that have a default value set. So maybe it would be a better idea to start adapting the code for doc generators now, when it's still a WIP. 

So if you'd like me to do the investigation on how to transform the source so that the doc gen will output the same thing as the manual writing, I'm more than happy to.
